### PR TITLE
Added a property to control value of NumericUpDown.

### PR DESCRIPTION
**Pull request by wunianqing**
**Describe the changes you have made to improve this project**

Please see the demo application. I have added a small demo in it.
The original `NumericUpDown` control will response every enter. This will cause a strange behavior when you binding value to a control like Slider. So I added a property to allow value effect after press enter. 

**Unit test**

I don't know how to test a custom control

**Additional context**

None

**Closed Issues**

None

### CODE REVIEW

File: src/MahApps.Metro.sln [hunk 1]
```diff
@@ -36,10 +36,10 @@ Global
...
 		{D9ACF0D0-A178-4C91-BD23-D013CEA763A0}.Release|Any CPU.Build.0 = Release|Any CPU
-		{40048BCF-D1C7-46CC-B781-05718BE25BFC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{40048BCF-D1C7-46CC-B781-05718BE25BFC}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{40048BCF-D1C7-46CC-B781-05718BE25BFC}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{40048BCF-D1C7-46CC-B781-05718BE25BFC}.Release|Any CPU.Build.0 = Release|Any CPU
+		{31D47FA4-9A93-4385-A326-76F189C02440}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{31D47FA4-9A93-4385-A326-76F189C02440}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{31D47FA4-9A93-4385-A326-76F189C02440}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{31D47FA4-9A93-4385-A326-76F189C02440}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
```
Comment (punker76) [inline src/MahApps.Metro.sln]: Are these changes really necessary?

File: src/MahApps.Metro/MahApps.Metro.csproj [hunk 2]
```diff
@@ -10,8 +10,8 @@
         <GenerateDocumentationFile>true</GenerateDocumentationFile>
     </PropertyGroup>
     <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp3.0'">
-        <Reference Include="System.ComponentModel.DataAnnotations"/>
-        <Reference Include="System.Configuration"/>
+        <Reference Include="System.ComponentModel.DataAnnotations" />
+        <Reference Include="System.Configuration" />
     </ItemGroup>
```
Comment (punker76) [inline src/MahApps.Metro/MahApps.Metro.csproj]: Are these changes really necessary?

File: src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+           TargetType="{x:Type Controls:NumericUpDown}"
+           BasedOn="{StaticResource {x:Type Controls:NumericUpDown}}">
+        <Style.Resources>
+            <Style x:Key="ChromelessButtonStyle" TargetType="{x:Type ButtonBase}"
+                   BasedOn="{StaticResource ChromelessButtonStyle}">
+                <Setter Property="Focusable" Value="False"/>
+            </Style>
+        </Style.Resources>
+    </Style>
+</ResourceDictionary>
```
Comment (punker76) [inline src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Why do we need a new style?

File: src/MahApps.Metro.sln (hunk 1 above)
Comment (wunianqing) [reply src/MahApps.Metro.sln]: Not necessary. It was added by VS automatically.

File: src/MahApps.Metro/MahApps.Metro.csproj (hunk 2 above)
Comment (wunianqing) [reply src/MahApps.Metro/MahApps.Metro.csproj]: No. I just pressed Ctrl+K, Ctrl+D

File: src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml (hunk 3 above)
Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Yes, we do need a new style. We need TextBox stay in focusing when click increase or decrease button. But we can make this shorter. I committed a new style in my repository.

Comment (punker76) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: > Yes, we do need a new style. We need TextBox stay in focusing when click increase or decrease button. But we can make this shorter. I committed a new style in my repository.

I think we can get the same result without creating/using a new style. We can prevent focusing of the up and down buttons permanently when we do this

image

image

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Yes, of course. I created a new style just want to keep the original behavior. If you think this is OK. Then we don't need a new style any more.

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Wait. I think I misunderstood you.

Comment (punker76) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: I will think about another DepProp, which will control this behavior and check into the PR branch here.

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: The purpose of my style is that when you set `ValueChangedOnLostFocus` to true, then you may want increase button and decrease button only effect the text, not the value. The value only changes when press enter.

Comment (punker76) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: No, that's not a good UX. The value should changed by using up and down buttons. Only if you type manually we should have this behavior.

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Yes, but this only happens when you focused on TextBox. If you are not focusing TextBox, then the buttons work.

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: And that's why I created a new style rather than change it directly on the default style. It is just an option.

Comment (punker76) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: No new style needed. Same can be done with e.g. a DepPropery UpDownButtosFocusable with default true to keep the current implementation. Then another property to handle your new requirement to change the value behind the user input manually by hitting enter.

Comment (wunianqing) [reply src/MahApps.Metro/Styles/Controls.NumericUpDown.xaml]: Cool. I am OK with that.
